xlators/storage/posix/src/posix-entry-ops.c

```diff
@@ -215,6 +220,28 @@ posix_lookup(call_frame_t *frame, xlator_t *this, loc_t *loc, dict_t *xdata)
     op_ret = -1;
     if (gf_uuid_is_null(loc->pargfid) || (loc->name == NULL)) {
         /* nameless lookup */
+        op_ret = op_errno = 0;
+        /* The gfid will be renamed to ".glusterfs/unlink" in case
+         * there are any open fds on the file in posix_unlink path.
+         * So client can request server to do nameless lookup with
+         * xdata = GF_UNLINKED_LOOKUP in ".glusterfs/unlink"
+         * dir if a client wants to know the status of the all open fds
+         * on the unlinked file. If the file still present in the
+         * ".glusterfs/unlink" dir then it indicates there still
+         * open fds present on the file and the file is still under
+         * unlink process */
+        ret = dict_get_uint32(xdata, GF_UNLINKED_LOOKUP, &lookup_unlink_dir);
+        if (!ret && lookup_unlink_dir) {
+            POSIX_GET_FILE_UNLINK_PATH(priv->base_path, loc->gfid, unlink_path);
+            ret = sys_lstat(unlink_path, &lstatbuf);
+            if (ret) {
+                op_ret = -1;
+                op_errno = ret;
+            }
+            goto nameless_lookup_unlink_dir_out;
+        }
+        /* If you are here means the call is for regular nameless
+         * lookup of a file */
```

Comment:
I'm not sure if this order is the right one. Checking first unlink path means that most of the lookups will require an additional system call to check for unlinks, which is not a common case. I think we should first try the regular approach, and if it fails with ENOENT, then check the unlink directory.

Reply:
sure...

Reply:
Why ? we should call `MAKE_INODE_HANDLE()` and check if it failed (`op_ret == -1`). If so, and `errno` is `ENOENT` , then we should check if the lookup request wants to check for unlinked inodes.